### PR TITLE
fix(ce-compound): guard validate-frontmatter.py on non-Claude platforms

### PR DESCRIPTION
**Pull request by tmchow**
## What & why

`ce-compound` and `ce-compound-refresh` invoked their parser-safety validator via
`python3 "${CLAUDE_SKILL_DIR:-.}/scripts/validate-frontmatter.py"`. `${CLAUDE_SKILL_DIR}`
is a **Claude-only content substitution**, so off-Claude the `:-.` fallback resolved
`./scripts/...` against the **project CWD**, the bundled script wasn't found, and the
silent-corruption guard was **silently skipped**. The plugin also ships as a *native
Codex plugin* (no converter runs), and `ce_platforms` is converter-only — so only
SKILL.md *content* can protect that path.

## Changes

- **Guard the validator** in both skills: `if [ -n "$CLAUDE_SKILL_DIR" ] && [ -f ... ]; then python3 ... else <manual checklist> fi`. The `else` branch runs an inline parser-safety checklist that mirrors the script's exact scope (top-level, unquoted/unstructured scalar values; delimiter lines whose stripped content is `---`), so the protection still fires off-Claude instead of vanishing — and never falsely reports success.
- **`skill-conventions` test detector** now recognizes existence/non-empty guards as graceful — **block-scoped** inside code fences (a second, *unguarded* bundled-script call in the same file is still flagged) and **file-wide** for prose mentions. Reuses the existing `partitionFencedCodeBlocks` state machine.
- **Docs** (`AGENTS.md` root + plugin): document the existence-guard pattern, the Claude-Code permission caveat, and the previously-undocumented fact that the **native Codex install ignores `ce_platforms`** (so content-level handling is the only cross-install protection).

## Scope / what's deferred

This is the portable, skill-independent slice of #943. The other skill #943 named, **`ce-worktree`**, is intentionally **not** touched here — its bundled `worktree-manager.sh` needs a re-architecture (into a portable isolation guardrail), tracked in **#946**, rather than a guard around the script. The broader bare-`scripts/...` invocation audit is **#944**.

## Testing

- `bun test`: only the 9 pre-existing environment failures (tool-detection / CLI-install, present on `main` too); **zero new**.
- `bun run release:validate`: in sync.

## Process

Design + code review across four rounds with an independent Codex reviewer (it caught a permission regression and a detector edge-case bug, both fixed); a `/simplify` pass deduped the guard-regex skeleton and replaced a per-occurrence rescan with a single precomputed set.

Refs #943 · related #944, #946

### CODE REVIEW

File: tests/skill-conventions.test.ts [hunk 1]
```diff
@@ -447,29 +447,107 @@ const PLATFORM_VAR_REGEX = /\$\{?((?:CLAUDE|CODEX)_[A-Z][A-Z0-9_]*)/g
...
+ * True when the occurrence on this line is gracefully handled *by the line
+ * itself*: a `${VAR:-...}` shell default, or a `[ -n/-z/-f/-e/-d "...$VAR..." ]`
+ * test (existence/non-empty guard).
+ */
 function isGracefulPlatformVarUse(line: string, variable: string): boolean {
-  const defaultForm = new RegExp(`\\$\\{${variable}:-`)
-  if (defaultForm.test(line)) return true
-  const guardForm = new RegExp(`\\[\\s*-[nz]\\s+"[^"]*\\$\\{?${variable}\\b[^"]*"\\s*\\]`)
-  return guardForm.test(line)
+  if (new RegExp(`\\$\\{${variable}:-`).test(line)) return true
```
[the remark this answers is not in the document]
Comment (tmchow) [reply tests/skill-conventions.test.ts]: Addressed in fa990c4c. The detector now excludes skill-directory vars (`CLAUDE_SKILL_DIR`, `CLAUDE_PLUGIN_ROOT`) from the `${VAR:-...}` graceful path — a `${CLAUDE_SKILL_DIR:-.}/scripts/...` bundled-script invocation is now reported as non-graceful, so a future skill can no longer silently reintroduce the #943 regression. Ordinary vars keep the shell-default fallback. Added regression guards (`does NOT treat ${CLAUDE_SKILL_DIR:-.} as graceful`, `a ${CLAUDE_SKILL_DIR:-.} bundled-script path IS reported`). ce-worktree still uses the legacy `:-.` form on this branch, so its acknowledgment reason now names it a known laggard tracked by #946/#948 (which removes the bundled script entirely).
